mapper/tedge_mapper/src/sm_c8y_mapper/tests.rs

```diff
@@ -292,31 +224,140 @@ async fn mapper_publishes_software_update_status_and_software_list_onto_c8y_topi
+    // Now the SM Mapper recovers and receives the response message and publishes it on `c8y/s/us`
+    // The subscriber that was waiting for the response on `c8/s/us` receives the response and validates it.
+
+    // Create a subscriber to receive messages on `tedge/commands/req/software/update` topic.
+    let mut sw_update_req_sub = get_subscriber(
+        "tedge/commands/req/software/update",
+        "mapper_publishes_software_update_request",
+    )
+    .await;
+
+    // Create a subscriber to receive messages on `"c8y/s/us` topic.
+    let mut sw_update_res_sub = get_subscriber(
+        "c8y/s/us",
+        "mapper_publishes_software_update_response_to_c8y_cloud",
+    )
+    .await;
+
+    // Start SM Mapper
+    let sm_mapper = start_sm_mapper().await?;
+
+    // Prepare and publish a software update response `successful`.
+    let json_response = r#"{
+        "id":"123",
+        "status":"successful",
+        "currentSoftwareList":[
+            {"type":"apt","modules":[
+                {"name":"m","url":"https://foobar.io/m.epl"}
+            ]}
+        ]}"#;
+
+    // Prepare and publish a software update smartrest request on `c8y/s/ds`.
+    let smartrest = r#"528, external_id,nodered,1.0.0::debian,,install"#;
+    let _ = publish(&Topic::new("c8y/s/ds").unwrap(), smartrest.to_string()).await;
+
+    let expected_update_list = r#"
+         "updateList": [
+            {
+                "type": "debian",
+                "modules": [
+                    {
+                        "name": "nodered",
+                        "version": "1.0.0",
+                        "action": "install"
+                    }
+                ]
+            }"#;
+
+    // Expect thin-edge json message on `tedge/commands/req/software/update` with expected payload.
+    match tokio::time::timeout(TEST_TIMEOUT_MS, sw_update_req_sub.next()).await {
+        Ok(Some(msg)) => {
+            dbg!(&msg.payload_str().unwrap());
+            if msg
+                .payload_str()
+                .unwrap()
+                .contains(&remove_whitespace(expected_update_list))
+            {
+                // Stop the SM Mapper
+                sm_mapper.abort();
+                //let _ = sm_mapper.await;
+                assert!(sm_mapper.await.unwrap_err().is_cancelled());
+
+                // Publish the response
+                let _ = publish(
+                    &Topic::new("tedge/commands/res/software/update").unwrap(),
+                    json_response.to_string(),
+                )
+                .await;
+            }
+        }
+        _ => {
+            panic!("No software update message received after a second.");
+        }
+    }
+
+    // Restart SM Mapper
+    let _sm_mapper = start_sm_mapper().await?;
+
+    let mut received_status_successful = false;
+
+    // Validate the response that is received on 'c8y/s/us'
+    for _ in 0..8 {
```

Comment:
Why 8?

Reply:
If you put the specific retry number, you should explain the reason as a comment.